src/filesystem.cpp

```diff
@@ -856,4 +853,207 @@ bool FileInputStream::read(byte *buffer, unsigned size)
     return false;
 }
 
+const std::string* LocalPath::editStringDirect() const
+{
+    // this function for compatibiltiy while converting to use LocalPath class.  TODO: phase out this function
```

Comment:
There's 79 usages of this method. It seems not really phased out. In case you consider it's important to stop using it, please, create a ticket, @mattw-mega

Reply:
Yes, I should have commented this one better - we're still using editStringDirect for now, in order to still access the std::string that is now wrapped up in this class, but has exactly the same content.  In order to hopefully make review easier, the std::string content is still the same now as it was before wrapping it up, so in review we can compare before and after and see that the contained string is still the same.   Later we can change it to std::wstring for windows which will match its use much, much better.   Ticket created: 16393.   But for now, as this ticket is already changing a huge number of lines (sorry about that), I've tried to limit the amount of change, and pass the old string to functions that didn't need to be converted to at this point, and so this function is still used quite a lot.    But I think the PR achieves the main aim, which is use a c++ type to represent the local paths, so it's much easier to track whether a std::string is containing utf8 or (for windows) utf16.   And hopefully minimizing the amount of lines that need to be reviewed to achieve that.   thanks!

Reply:
Thanks, I appreciate you care about the amount of lines in the diff. The PR is, with no doubt, a step in the right direction, it's great you take the time to improve the readability and maintenance of the code :+1: